src/commands.ts

```diff
@@ -79,6 +81,9 @@ export class Commands {
         throw new Error("Failed to get authenticated user")
       }
       await vscode.commands.executeCommand("setContext", "coder.authenticated", true)
+      if (user.roles.find((role) => role.name === "owner")) {
+        await vscode.commands.executeCommand("setContext", "coder.isOwner", true)
+      }
```

Comment:
Why do we need to check for the owner role? We prefer to do capability checks. You can send a payload to `https://dev.coder.com/api/v2/authcheck` to see if you can do something. Eg:
```json
{
   "checks":{
      "readAllUsers":{
         "object":{
            "resource_type":"user"
         },
         "action":"read"
      },
   },
}
```

Is this just to see if we have arbitrary execution of all workspaces?

Reply:
I think this is ok for now, we but we might want to change this soon, especially with https://github.com/coder/coder/pull/6875